### ISSUE TITLE
problem: [/addcoin] user not conveniently informed of errors 

### ISSUE THREAD

**Issue by chamidu-supeshala** [Blockrazor/blockrazor#1549]
Problem: for add coin wizard, the more of validations happens on the server side, only after completely submitting the form. 
So when there is validation issue in an intermediary step, it wont focused, user has to go back step by step searching where's the issue

Solution: 
1. introduce step by step validations with server. 
(currently there are only client side step validations, which only checks whether some fields non-empty, and not checks their values are valid or within the range etc.)

or

2. after wizard form submitted, if there is validation issue in intermediary step, automatically navigate to that step and highlight the fields which has issues

or

3. at least  highlight the step headers, where the validation issues exist. so user can manually navigate to that step and correct

(If step by step dynamic validation not introducing, some validations like "currency symbol cannot exceed 5 characters" need to be also happen in the client side, otherwise very annoying to go back in the final moment. It is ok if going with #1 solution)

**Comment by gazhayes** [comment]
Whoever does this please pay very careful attention to #1556

**Comment by janat08** [comment]
@blockrazorbot claim, closing the one I opened previously, #868, following it's solution.

**Comment by janat08** [comment]
btw autoForm doesn't depend on allow/deny, creating it's own method that follows the method. I'll likely look for inspiration there.

**Comment by gazhayes** [comment]
Oh I was under the impression that autoform required using allow/deny.

I actually don't care how we do validation, as long as it's consistent. I wrote #1556 because people are using different stuff all over the place even after we've settled on simpl-schema + validated-method months ago. But if you have a simpler way to solve the problem that's fine, just make sure you comment it extremely well so that it can be reproduced by everyone else.

**Comment by janat08** [comment]
I simply saw this: https://github.com/aldeed/meteor-autoform/blob/devel/formTypes/method.js. @anbud?

**Comment by gazhayes** [comment]
Ah yes autoform itself doesn't. But autoform doesn't offer much without collection2, and collection2 needs client side mongo writes to work with allow/deny and simpl-schema rules.

**Comment by janat08** [comment]
https://github.com/aldeed/meteor-autoform#method
```
method
Will call the server method with the name you specify in the meteormethod attribute. Passes a single argument, doc, which is the document resulting from the form submission.

You may optionally specify a DDP Connection in the ddp attribute. If you do, the method will be called using the DDP connection provided.

The method is not called until doc is valid on the client.
```

**Comment by gazhayes** [comment]
So you want to use this with validated-method and simplschema? Or are you proposing to completely rework everything?

**Comment by janat08** [comment]
I'd still use the validated method to validate on server as that doesn't happen with custom method param, but in integrating autoform I would remove current form logic in favour of autoform to given control to schema which would involve passing labels/select options/if field is to appear. 

In way of step validation I think I can only use AutoForm.validateField(formId, fieldName, skipEmpty) on every field in view.

**Comment by janat08** [comment]
PS: the guide talked about deciding if a field should appear, I realize now that there're no apparent directives in schema which would let autoform know that, so I might not do that.

**Comment by gazhayes** [comment]
Oh, I should have mentioned this earlier, but the first solution is the only one that really solves the actual issue (the real issue here is that validation is annoying for users). So validation is needed at each step rather than letting users go through the whole form before validation.

**Comment by gazhayes** [comment]
It's best to open fresh issues instead of reopen old ones, keep everything very tight and specific in regards to the problem (this issue). Then again, if they aren't big changes, you can just send a number of small commits to address each smaller problem and tag the pull request with this master issue.

**Comment by gazhayes** [comment]
I'm pretty sure autoform doesn't really help here, but if you think it makes it easier for you to solve the issue then I have no problem with you using it and seeing what happens.

**Comment by anbud** [comment]
@janat08, I haven't used autoform in ages, I can't really help here. If you get stuck somewhere though, let me know and I'll take a look.

**Comment by janat08** [comment]
Fine i'll just fish out the code behind AutoForm.validateField(formId, fieldName, skipEmpty)

**Comment by anbud** [comment]
@janat08, I've talked to @gazhayes, and, to speed things up, I'll take this over until you return if you don't mind. :)

**Comment by anbud** [comment]
In #1660, each field is validated using the simple-schema we've already defined. Value is validated within the current context. There is some UI stuff to be taken care of here, to make the validation look nice, but the basic functionality is here, so the UI stuff can go into another issue probably.

Now, I really think we should go over the schema again, as, when I was testing it, it seems that some fields that should be required aren't required at all.

@blockrazorbot label "done"
@blockrazorbot remove "in progress"

**Comment by janat08** [comment]
did you check that custom validation functions actually access other fields?

**Comment by anbud** [comment]
@janat08, I did, custom functions have access to everything as I've extended the context.

If you think that's causing issues with required fields, I'll take a deeper look. :)
